source/grid/grid_in.cc

```diff
@@ -493,6 +498,29 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
 }
 
 
+
+template <int dim, int spacedim>
+void
+GridIn<dim, spacedim>::read_vtu(std::istream &in)
+{
+  namespace pt = boost::property_tree;
+  pt::ptree tree;
+  pt::read_xml(in, tree);
+  auto section = tree.get_optional<std::string>("VTKFile.dealiiData");
+
+  AssertThrow(section,
+              ExcMessage(
+                "While reading VTU file, failed to find dealiiData section"));
```

Comment:
Can you make the error message a bit more descriptive? In particular, we sould get the point across that we can only read in vtus that were generated by us.